flang/lib/Semantics/check-declarations.cpp

```diff
@@ -963,7 +963,17 @@ void CheckHelper::CheckObjectEntity(
         "'%s' is a data object and may not be EXTERNAL"_err_en_US,
         symbol.name());
   }
-
+  if (symbol.test(Symbol::Flag::CrayPointee)) {
+    // NB, IsSaved was too smart here.
+    if (details.init()) {
+      messages_.Say(
+          "Cray pointee '%s' may not be initialized"_err_en_US, symbol.name());
+    } else if (symbol.attrs().test(Attr::SAVE)) {
```

Comment:
The `else` here means that at most one message will issue.  Can't both be useful?

Reply:
You are correct, I was thinking that anytime the statement was initialize you would get an extra error message about having the save attribute, and that being distracting / misleading. Now that I think about it again and more generally, we would want to show both when the variable has an explicit save attribute and was initialized.